### PR TITLE
Allow for arbitrary redirect URIs.

### LINKED ISSUE #156: eBay API Redirect URI
eBay's API has its own Redirect URL or RuName, as they call it, and that is expected in place of the https://script.google.com/macros/d/{SCRIPT ID}/usercallback URL. Is there a workaround for this issue or will eBay's API not work within the scope of a Google script?

### PR DESCRIPTION
**Pull request by erickoledadevrel**
Needed for the OAuth provider eBay, which requires you to register redirect URI pairs ahead of time and reference the pair by ID. Fixes #156.

### CODE REVIEW

**Comment by erickoledadevrel** [comment]
Ping.

File: src/OAuth2.js [hunk 1]
```diff
@@ -59,11 +59,13 @@ function createService(serviceName) {
...
- * @param {string} scriptId The script ID of your script, which can be found in
- *     the Script Editor UI under "File > Project properties".
+ * @param {string} [optScriptId] The script ID of your script, which can be
+ *     found in the Script Editor UI under "File > Project properties". Defaults
+ *     to the script ID of the script being executed.
  * @return {string} The redirect URI.
  */
-function getRedirectUri(scriptId) {
+function getRedirectUri(optScriptId) {
+  var scriptId = optScriptId || eval('Script' + 'App').getScriptId();
```
Comment (sqrrrl) [inline src/OAuth2.js]: The eval('Script' + 'App') call here is weird -- why necessary to eval and why the concatenation of constants? Shouldn't this just be ScriptApp.getScriptId()?

Comment (sqrrrl) [reply src/OAuth2.js]: BTW - If this was an attempt to thwart oauth scope detection, it looks like calling ScriptApp.getScriptId() directly doesn't actually trigger an auth requirement and is safe to use.

Comment (erickoledadevrel) [reply src/OAuth2.js]: It was a workaround to scope issues, but it looks like the problem it was working around was fixed in mid-2017, so no longer needed. Removed.

**Pull request merged**